ehttp/src/web.rs

```diff
@@ -9,11 +9,21 @@ use crate::{Request, Response};
 /// NOTE: `Ok(…)` is returned on network error.
 /// `Err` is only for failure to use the fetch API.
 pub async fn fetch_async(request: &Request) -> crate::Result<Response> {
-    fetch_jsvalue(request).await.map_err(string_from_js_value)
+    fetch_jsvalue(request).await.map_err(fetch_error_to_string)
 }
 
-pub(crate) fn string_from_js_value(value: JsValue) -> String {
-    value.as_string().unwrap_or_else(|| format!("{:#?}", value))
+/// This should only be used to handle opaque exceptions thrown by the `fetch` call.
+pub(crate) fn fetch_error_to_string(value: JsValue) -> String {
```

Comment:
nit: `fetch_error_to_string` sounds like a command ("fetch this error to a string please"). `string_from_fetch_error` avoids that.